pkg/operator/ceph/cluster/osd/osd.go

```diff
@@ -600,14 +600,17 @@ func (c *Cluster) startOSDDaemonsOnPVC(pvcName string, config *provisionConfig,
 				logger.Infof("deployment for osd %d already exists. updating if needed", osd.ID)
 				if err = updateDeploymentAndWait(c.context, c.clusterInfo, dp, opconfig.OsdType, strconv.Itoa(osd.ID), c.spec.SkipUpgradeChecks, c.spec.ContinueUpgradeAfterChecksEvenIfNotHealthy); err != nil {
 					logger.Errorf("failed to update osd deployment %d. %v", osd.ID, err)
+					// TODO: should orch status be updated here also?
```

Comment:
Yes.

Reply:
That makes sense. Is it worthwhile to consider whether we should log these to a `CephCluster` status item?

Reply:
Actually we should really change this behavior. This was put in place before we had the ability to re-queue a failed reconcile with the controller runtime. We should be re-queuing the reconcile if any of the OSDs fail to update, otherwise some OSDs may not be updated and we wouldn't retry until the next reconcile happens to be triggered.